lib/salesforce_bulk/client.rb

```diff
@@ -38,6 +41,10 @@ def initialize(options={})
     end
 
     def authenticate
+      # Clear session attributes just in case client already had a session
+      @session_id = nil
+      self.instance_host = nil
```

Comment:
Without this, calling `authenticate` a second time would fail because of this code branch: https://github.com/controlshift/salesforce_bulk/blob/re_authenticate_support/lib/salesforce_bulk/client.rb#L214